src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -559,5 +621,529 @@ static void Main()
+  IL_0038:  stloc.0
+  IL_0039:  dup
+  IL_003a:  ldfld      ""long System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Item5""
+  IL_003f:  stloc.0
+  IL_0040:  dup
+  IL_0041:  ldfld      ""long System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Item6""
+  IL_0046:  stloc.0
+  IL_0047:  dup
+  IL_0048:  ldfld      ""long System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Item7""
+  IL_004d:  stloc.0
+  IL_004e:  dup
+  IL_004f:  ldfld      ""(long, long, int) System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Rest""
+  IL_0054:  ldfld      ""long System.ValueTuple<long, long, int>.Item1""
+  IL_0059:  stloc.0
+  IL_005a:  dup
+  IL_005b:  ldfld      ""(long, long, int) System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Rest""
+  IL_0060:  ldfld      ""long System.ValueTuple<long, long, int>.Item2""
+  IL_0065:  stloc.0
+  IL_0066:  ldfld      ""(long, long, int) System.ValueTuple<long, long, long, long, long, long, long, (long, long, int)>.Rest""
+  IL_006b:  ldfld      ""int System.ValueTuple<long, long, int>.Item3""
+  IL_0070:  stloc.1
+  IL_0071:  ldloc.0
+  IL_0072:  box        ""long""
+  IL_0077:  ldstr      "" ""
+  IL_007c:  ldloc.1
+  IL_007d:  box        ""int""
+  IL_0082:  call       ""string string.Concat(object, object, object)""
+  IL_0087:  call       ""void System.Console.WriteLine(string)""
+  IL_008c:  ret
+}
+");
+        }
+
+        [Fact(Skip = "PROTOTYPE(tuples)")]
+        public void DeconstructLongTuple2()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        long x;
+        int y;
+
+        (x, x, x, x, x, x, x, x, x, y) = (1, 1, 1, 1, 1, 1, 1, 1, 4, (byte)2);
+        System.Console.WriteLine(x + "" "" + y);
+    }
+}
+";
+
+            // issue with return type
+            var comp = CompileAndVerify(source, expectedOutput: "4 2", additionalRefs: new[] { ValueTupleRef, SystemRuntimeFacadeRef }, parseOptions: TestOptions.Regular.WithTuplesFeature());
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void DeconstructTypelessTuple()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        string x = ""goodbye"";
+        string y;
+
+        (x, y) = (null, ""hello"");
+        System.Console.WriteLine($""{x}{y}"");
+    }
+}
+";
+
+            var comp = CompileAndVerify(source, expectedOutput: "hello", additionalRefs: new[] { ValueTupleRef, SystemRuntimeFacadeRef }, parseOptions: TestOptions.Regular.WithTuplesFeature());
+            comp.VerifyDiagnostics();
+            comp.VerifyIL("C.Main", @"
+{
+  // Code size       48 (0x30)
+  .maxstack  3
+  .locals init (string V_0, //x
+                string V_1) //y
+  IL_0000:  ldstr      ""goodbye""
+  IL_0005:  stloc.0
```

Comment:
A bit surprising that IL_0000, IL_0005 are necessary. Will need to take a look when get to optimizations.